packages/backend/src/handlers/actions/quests/createCompletion/createCompletion.ts

```diff
@@ -0,0 +1,67 @@
+import { CreateQuestCompletionInput, CreateQuestCompletionOutput,
+  Quest_Completion_Insert_Input,
+  QuestRepetition_Enum,
+  QuestStatus_Enum, } from '../../../../lib/autogen/hasura-sdk';
+import { client } from '../../../../lib/hasuraClient';
+
+export async function createCompletion(
+  playerId: string,
+  questCompletion: CreateQuestCompletionInput,
+): Promise<CreateQuestCompletionOutput> {
+  if (!questCompletion.submission_link && !questCompletion.submission_text) {
+    throw new Error('Must provide at least a submission link or text');
+  }
+
+  const { quest_by_pk: quest } = await client.GetQuestById({ quest_id: questCompletion.quest_id });
+  if (!quest) {
+    throw new Error('Quest not found');
+  }
+
+  if (quest.status !== QuestStatus_Enum.Open) {
+    throw new Error('Quest must be open');
+  }
+
+  // Personal or unique, check if not already done by player
+  if (quest.repetition === QuestRepetition_Enum.Unique || quest.repetition === QuestRepetition_Enum.Personal) {
+    const { quest_completion: existingQuestCompletions } = await client.GetQuestCompletions({
+      player_id: playerId,
+      quest_id: questCompletion.quest_id,
+    });
+    if(existingQuestCompletions.length > 0) {
+      throw new Error('Player already completed this personal/unique quest');
+    }
+  }
+
+  // Recurring, check if not already done by player within cooldown
+  if (quest.repetition === QuestRepetition_Enum.Recurring && quest.cooldown) {
+    const { quest_completion: existingQuestCompletions } = await client.GetLastQuestCompletionForPlayer({
+      player_id: playerId,
+      quest_id: quest.id,
+    });
+    if(existingQuestCompletions.length > 0) {
+      const existingQuestCompletion = existingQuestCompletions[0];
+      const submittedAt = new Date(existingQuestCompletion.submitted_at);
+      const now = new Date();
+      const diff = +now - +submittedAt;
```

Comment:
Haven't seen this syntax before, what does the `+`  do?

Reply:
convert the Date to a timestamp number